scipy/optimize/_quasi_newton_approx.py

```diff
@@ -0,0 +1,261 @@
+"""quasi-Newton Hessian approximations"""
+
+from __future__ import division, print_function, absolute_import
+import numpy as np
+from numpy.linalg import norm
+from scipy.linalg import get_blas_funcs
+
+
+class QuasiNewtonApprox(object):
+    """Quasi-Newton approximation."""
+
+    _syr = get_blas_funcs('syr', dtype='d')  # Symetric rank 1 update
+    _syr2 = get_blas_funcs('syr2', dtype='d')  # Symetric rank 2 update
+    _symv = get_blas_funcs('symv', dtype='d')  # Symetric matrix-vector product
+
+    def instanciate_matrix(self, delta_x, delta_grad, approx_type='hess'):
+        """Instanciate unscaled matrix."""
+        n = len(delta_x)
+        self.approx_type = approx_type
+        if approx_type not in ('hess', 'inv_hess'):
+            raise ValueError("Unknown approx_type.")
+        # Create matrix
+        if self.approx_type == 'hess':
+            self.B = np.eye(n, dtype=float)
+        else:
+            self.H = np.eye(n, dtype=float)
+
+    def scale_matrix(self, delta_x, delta_grad):
```

Comment:
Method `update` should be implemented as abstract (raise NotImpementedError), with an explanation how to implement a new strategy.

Reply:
I agree with your two comments: I should include an abstract ``update`` method and it would be good to have the scale inside the update (rather than in a separate method).

I will just correct the user example to make it coherent with my previous comment:
```
# User side
bfgs = BFGS(exception_strategy='damped_bfgs', min_curvature=1e-10, [other user options])
# Inside `minimize` call
bfgs.initialize(n, 'hess')
# Use by the solver:
bfgs.update(delta_x, delta_grad)
```